### PR TITLE
Add power module

### PR DESCRIPTION
**Pull request by kbeckmann**

### CODE REVIEW

File: app/drivers/adc.c [hunk 1]
```diff
@@ -3,11 +3,9 @@
 #include "drivers/gpio.h"
 #include "drivers/adc.h"
 
-#define NUM_OF_CHANNELS 2
-
 ADC_HandleTypeDef adc_handle;
 static bool m_initialized;
-static uint16_t m_adc_values[NUM_OF_CHANNELS];
+static uint16_t m_adc_values[ADC_NUM_OF_CHANNELS];
```
Comment (arturo182) [inline app/drivers/adc.c]: I feel like this should be NUM_OF_ADC_CHANNELS to match the other enum NUM_* stuff

File: app/power.c [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+	r = mcp4018t_set_value(value);
+	ERR_CHECK(r);
+
+	m_ldo_voltage = millivolt;
+
+	return r;
+}
+
+void adc_conversion_ready_handler(const uint16_t adc_values[ADC_NUM_OF_CHANNELS])
```
Comment (arturo182) [inline app/power.c]: Not uint16_t * ?

File: app/power.c [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+{
+	BaseType_t higher_priority_task_woken = pdFALSE;
+
+	xQueueSendFromISR(power_queue_handle, adc_values, &higher_priority_task_woken);
+	portYIELD_FROM_ISR(higher_priority_task_woken);
+}
+
+void power_task(void *p_arg)
```
Comment (arturo182) [inline app/power.c]: static ?

File: app/power.h [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+#include <stdbool.h>
+#include "errors.h"
+
+#define EPOWER_NO_INIT			(POWER_BASE + 0)
+#define EPOWER_TASK_CREATE		(POWER_BASE + 1)
+#define EPOWER_QUEUE_CREATE		(POWER_BASE + 2)
+#define EPOWER_VOLTAGE_RANGE	(POWER_BASE + 3)
+
+void power_dut_set_enabled(bool enabled);
+err_t power_dut_ldo_set(uint32_t millivolt);
```
Comment (arturo182) [inline app/power.h]: Maybe add getters for these two as well, for future ;)

File: app/drivers/adc.c (hunk 1 above)
Comment (kbeckmann) [reply app/drivers/adc.c]: Just a bit worried about name clashes when there isn’t a prefix

Comment (kbeckmann) [reply app/drivers/adc.c]: Just a bit worried about name clashes when there isn’t a prefix

Comment (kbeckmann) [reply app/drivers/adc.c]: Just a bit worried about name clashes when there isn’t a prefix

Comment (kbeckmann) [reply app/drivers/adc.c]: Just a bit worried about name clashes when there isn’t a prefix

File: app/power.c (hunk 2 above)
Comment (kbeckmann) [reply app/power.c]: Which line?

File: app/power.c (hunk 3 above)
Comment (kbeckmann) [reply app/power.c]: Sure

File: app/power.h (hunk 4 above)
Comment (kbeckmann) [reply app/power.h]: Sure

File: app/power.c (hunk 2 above)
Comment (arturo182) [reply app/power.c]: The comments are for the line that is just above the content so #L62

Comment (kbeckmann) [reply app/power.c]: Ah. No the idea here is that by doing like this i don't need to pass a pointer + the number of values. Fewer things to do in the interrupt handler. The consumer of the API also understands how many items there are, so there can't be any confusion.

The generated assembly when doing like this or using a pointer is the same.

You can see the change in the API in the other commit in this PR. (https://github.com/xil-se/xildebug_sw/pull/24/files#diff-6f418b016d24c723ec049507e7846810R12)

**Comment by kbeckmann** [comment]
Added getters and so on.

**Pull request merged**